### PR TITLE
Add dimension_type and sort_index to dimension_description

### PR DESCRIPTION
**Pull request by ewelinagr**
There are 2 new columns in dimension description introduced in [transmart 17.1-HYVE-5.8-RC1](https://github.com/thehyve/transmart-core/releases/tag/v17.1-hyve-5.8-rc.1), related to the sample based querying support:
- dimension description - Indicates whether the dimension represents subjects or observation attributes. [SUBJECT, ATTRIBUTE],
- sort index - Specifies a relative order between dimensions.

Issue description: [TMT-858](https://jira.thehyve.nl/browse/TMT-858).
Related conversion pipline PR: https://github.com/thehyve/pmc-conversion/pull/71

### CODE REVIEW

**Comment by ewelinagr** [comment]
Fixing tests in progress...

File: tmtk/clinical/modifier.py [hunk 1]
```diff
@@ -55,7 +55,9 @@ def create_df(self):
                               'modifier_path': ['\Missing Value', '\Sample_id'],
                               'modifier_cd': ['MISSVAL', 'SAMPLE_ID'],
                               'name_char': ['Missing Value','Sample identifier'],
-                              'Data Type': ['CATEGORICAL', 'CATEGORICAL']}
+                              'Data Type': ['CATEGORICAL', 'CATEGORICAL'],
```
Comment (alepev) [inline tmtk/clinical/modifier.py]: does it matter that here "Data **T**ype" is spelled like this and not "Data **t**ype" as in previous code above (e.g. modifiers_file.txt)?

Comment (alepev) [reply tmtk/clinical/modifier.py]: Although it is spelled this way in the pmc-conversion repo (https://github.com/thehyve/pmc-conversion/pull/71/files)

File: tmtk/toolbox/skinny_loader/i2b2metadata/dimension_descriptions.py [hunk 2]
```diff
@@ -33,8 +39,13 @@ def adapt_df_row(x):
             if not any(modifier_row):
                 return
 
+            size = len(x)
             self.df.loc[modifier_row, 'modifier_code'] = x[1]
             self.df.loc[modifier_row, 'value_type'] = 'T' if x[3] == 'CATEGORICAL' else 'N'
+            if size > 4:
```
Comment (Spayralbe) [inline tmtk/toolbox/skinny_loader/i2b2metadata/dimension_descriptions.py]: @ewelinagr Is it necessary to make these conditional?

Comment (ewelinagr) [reply tmtk/toolbox/skinny_loader/i2b2metadata/dimension_descriptions.py]: dimension_type and sort index are optional, so I wanted not to make them required here

**Pull request merged**